### PR TITLE
chore: RHDEVDOCS-5317 update to OCP 4.12 as...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

chore: [RHDEVDOCS-5317](https://issues.redhat.com//browse/RHDEVDOCS-5317) update to OCP 4.12 as the recommended version and make sure all links have 'link:' prefix so they render correctly

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by max-cx** [comment]
@nickboldt, to answer your question, yes we need a downstream merge request. I'll open one just now.

File: antora.yml
Comment (max-cx) [inline antora.yml:27]: @themr0c, your the Antora expert, do you happen to remember from the top of your head if we can do something like this in the antora.yml file?

File: antora.yml
Comment (max-cx) [inline antora.yml:89]: Waiting for Fabrice's confirmation ^.

File: antora.yml
Comment (nickboldt) [reply antora.yml:27]: check out line 52:

```
    link-manage-compute-resources-container: "link:https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[{kubernetes} - Resource management for pods and containers]"
```

this resolves to line 49's content:
```
    kubernetes: Kubernetes
```

Comment (max-cx) [reply antora.yml:27]: Alright. In the worst case, we get a failed GH action test.

File: antora.yml
Comment (max-cx) [inline antora.yml]: I'm checking the Cabal call doc for 7.64/3.6, and it reads `0.20`, although I'm not sure what's the correct value in the X.Y.Z format, like this?
```suggestion
    devworkspace-operator-version-patch: "0.20.0"
```

Comment (nickboldt) [reply antora.yml]: 3.5 uses 0.19; 3.6 uses 0.20. I wouldn't change this until you're updating the 3.6 docs.

File: antora.yml
Comment (themr0c) [reply antora.yml:27]: I believe it should be OK. But please test it to assert that it works.

File: antora.yml
Comment (max-cx) [reply antora.yml]: @nickboldt, thanks, we already branched out to a separate branch for 3.5. This repo's main is past 3.5 now. So we're committing this suggestion, OK?

Comment (nickboldt) [reply antora.yml]: yes, the correct version is 
```
devworkspace-operator.v0.20.0
```

**Comment by max-cx** [comment]
Merge to `main`.
Cherry-pick to `7.64.x` and `7.65.x` (and also `7.66.x` if the branch is created before the merge).

**Review by max-cx — APPROVED** [review]

**Pull request merged**
